internal/stackdiag.go

```diff
@@ -476,3 +496,24 @@ func extractEsInfo(typ string, ns string, resourceInfo *resource.Info) (bool, st
 
 	return isTLS, esName, nil
 }
+
+// determineESCredentialsSecret returns the name of the secret containing the Elasticsearch credentials attempting
+// first to use the "elastic" user secret, and then attempting to use the newer "elastic-internal-diagnostics" user
+// and returning an error if either cannot be found, or the "elastic-internal-diagnostics" user does not have the
+// expected key.
+func determineESCredentialsSecret(k *Kubectl, ns, esName string) (secretName, secretKey string, err error) {
+	elasticSecretName := fmt.Sprintf("%s-es-elastic-user", esName)
```

Comment:
Should it not be the other way around? Default to using the dedicated diagnostic user. If not availalbe try to fallback to the class superuser?

Reply:
You marked this as resolved but the code still goes with the `elastic` user first? Did something change your mind?

Reply:
🤦 I failed to push this commit last night after adjusting the order @pebrc. I'm adjusting the other things you noted now..